layers/+lang/emacs-lisp/packages.el

```diff
@@ -61,6 +61,25 @@
           "mhh" 'elisp-slime-nav-describe-elisp-thing-at-point)))))
 
 (defun emacs-lisp/init-emacs-lisp ()
+
+  (defun spacemacs//warn-on-old-byte-compiled-file ()
+    "Warn if it's emacs-lisp-mode and compiled file exists and is
+older than the current file."
+    (when (and (eq major-mode 'emacs-lisp-mode)
+               (file-newer-than-file-p
+                (buffer-file-name)
+                (byte-compile-dest-file buffer-file-name)))
```

Comment:
Does that work if there's no byte compiled file?  IOW, does `file-newer-than-file-p` cope with non-existing files?